File: leetcode/easy/remove-digit-from-number-to-maximize-result.ts
// https://leetcode.com/problems/remove-digit-from-number-to-maximize-result/

// MINE..
/*
function removeDigit(num: string, digit: string): string {
    const A = []
    for(let i=0; i<num.length; ++i) {
        if(num[i] === digit) {
            const digits = num.split('')
            digits.splice(i, 1)
            A.push(parseInt(digits.join('')))
        }
    }
    console.log({A})
    return String(Math.max(...A))
};
*/

// ANSWER
function removeDigit(num: string, digit: string): string {
  for (let i = 0; i < num.length - 1; ++i)
    if (num[i] === digit && num[i] < num[i + 1]) return num.substring(0, i) + num.substring(i + 1)
  for (let i = num.length - 1; i >= 0; --i)
    if (num[i] === digit) return num.substring(0, i) + num.substring(i + 1)
  return ''
}

File: leetcode/easy/remove-digit-from-number-to-maximize-result.py
# https://leetcode.com/problems/remove-digit-from-number-to-maximize-result/

class Solution:
    def removeDigit(self, num: str, digit: str) -> str:
        for i in range(len(num)-1):
            if num[i] == digit and num[i] < num[i+1]:
                return num[:i] + num[i+1:]
        last = num.rfind(digit)
        return num[:last] + num[last+1:]
